xbuild/src/command/build.rs

```diff
@@ -261,9 +261,67 @@ pub fn build(env: &BuildEnv) -> Result<()> {
             if has_lib {
                 for target in env.target().compile_targets() {
                     let arch_dir = platform_dir.join(target.arch().to_string());
-                    let lib =
-                        env.cargo_artefact(&arch_dir.join("cargo"), target, CrateType::Cdylib)?;
+                    let cargo_dir = arch_dir.join("cargo");
+                    let lib = env.cargo_artefact(&cargo_dir, target, CrateType::Cdylib)?;
                     apk.add_lib(target.android_abi(), &lib)?;
+
+                    let ndk = env.android_ndk();
```

Comment:
@dvc94ch I checked this and `has_lib` is false, that's why this bit of code never runs.

Reply:
There's a `libapp.so` but I don't see what `main.rs` compiled into...

Reply:
fyi adding the template to ci https://github.com/cloudpeers/xbuild/pull/71

Reply:
Okay, built it and `has_lib` is true, I see `libflutter.so` being added but it's not in the required dylibs:

```console
$ llvm-readelf --needed-libs target/x/release/android/arm64/cargo/aarch64-linux-android/release/libfoo.so
NeededLibraries [
  libc.so
  libdl.so
]
```

Reply:
ah, makes sense. it's probably dlopened by the flutter java code. sorry for the false alarm. In that case this is resolved.

Reply:
> you should not have to install dart. it is already prepackaged and shipped. what os are you trying this on?

Arch linux. It did download things but I think `dart` wasn't made available to the path properly? Hopefully the CI will catch it and allow you to debug, as it wasn't an issue in `helloworld` with just `main.rs`.

> fyi adding the template to ci #71

Thanks, nice! I'll rebase on that when it's merged so that we can be doubly sure that this PR is not breaking Flutter.

> ah, makes sense. it's probably dlopened by the flutter java code. sorry for the false alarm. In that case this is resolved.

Yeah, I guess it must be dlopened then. No worries, better safe than sorry!

Reply:
can you let me know when that happened? was it during the build or when trying to run?

Reply:
Yup, it's `FfiGen` in `build.rs`:

<details>

```
   Compiling foo v0.1.0 (xbuild/examples/foo)
error: failed to run custom build command for `foo v0.1.0 (xbuild/examples/foo)`

Caused by:
  process didn't exit successfully: `xbuild/examples/foo/target/x/release/android/arm64/cargo/release/build/foo-09500578752ec7e1/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=xbuild/examples/foo/api.rsh

  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: dart not installed

  Caused by:
      No such file or directory (os error 2)

  Stack backtrace:
     0: <E as anyhow::context::ext::StdError>::ext_context
     1: anyhow::context::<impl anyhow::Context<T,E> for core::result::Result<T,E>>::context::{{closure}}
     2: core::result::Result<T,E>::map_err
     3: anyhow::context::<impl anyhow::Context<T,E> for core::result::Result<T,E>>::context
     4: ffi_gen::FfiGen::generate_dart
     5: build_script_build::main
     6: core::ops::function::FnOnce::call_once
     7: std::sys_common::backtrace::__rust_begin_short_backtrace
     8: std::rt::lang_start::{{closure}}
     9: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
               at /rustc/90ca44752a79dd414d9a0ccf7a74533a99080988/library/core/src/ops/function.rs:280:13
    10: std::panicking::try::do_call
               at /rustc/90ca44752a79dd414d9a0ccf7a74533a99080988/library/std/src/panicking.rs:492:40
    11: std::panicking::try
               at /rustc/90ca44752a79dd414d9a0ccf7a74533a99080988/library/std/src/panicking.rs:456:19
    12: std::panic::catch_unwind
               at /rustc/90ca44752a79dd414d9a0ccf7a74533a99080988/library/std/src/panic.rs:137:14
    13: std::rt::lang_start_internal::{{closure}}
               at /rustc/90ca44752a79dd414d9a0ccf7a74533a99080988/library/std/src/rt.rs:128:48
    14: std::panicking::try::do_call
               at /rustc/90ca44752a79dd414d9a0ccf7a74533a99080988/library/std/src/panicking.rs:492:40
    15: std::panicking::try
               at /rustc/90ca44752a79dd414d9a0ccf7a74533a99080988/library/std/src/panicking.rs:456:19
    16: std::panic::catch_unwind
               at /rustc/90ca44752a79dd414d9a0ccf7a74533a99080988/library/std/src/panic.rs:137:14
    17: std::rt::lang_start_internal
               at /rustc/90ca44752a79dd414d9a0ccf7a74533a99080988/library/std/src/rt.rs:128:20
    18: std::rt::lang_start
    19: main
    20: __libc_start_call_main
    21: __libc_start_main@GLIBC_2.2.5
    22: _start', build.rs:12:10
  stack backtrace:
     0: rust_begin_unwind
               at /rustc/90ca44752a79dd414d9a0ccf7a74533a99080988/library/std/src/panicking.rs:584:5
     1: core::panicking::panic_fmt
               at /rustc/90ca44752a79dd414d9a0ccf7a74533a99080988/library/core/src/panicking.rs:142:14
     2: core::result::unwrap_failed
               at /rustc/90ca44752a79dd414d9a0ccf7a74533a99080988/library/core/src/result.rs:1785:5
     3: core::result::Result<T,E>::unwrap
     4: build_script_build::main
     5: core::ops::function::FnOnce::call_once
  note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

</details>